client/post-editor/term-token-field/index.jsx

```diff
@@ -96,8 +99,13 @@ TermTokenField.propTypes = {
 export default connect( ( state, props ) => {
 	const siteId = getSelectedSiteId( state );
 
+	const postType = getEditedPostValue( state, siteId, getEditorPostId( state ), 'type' );
+	const taxonomies = getPostTypeTaxonomies( state, siteId, postType ) || [];
+	const taxonomy = taxonomies.filter( tax => tax.name === props.taxonomyName )[ 0 ];
```

Comment:
We could probably benefit from a `getPostTypeTaxonomy` selector. Also kinda wish I had preserved the object structure for easier lookup to avoid a filter like you've had to do here. All the same, [`_.find`](https://lodash.com/docs#find) would be a better choice because it halts iteration once it finds the matching entry (unlike `Array#filter`).